apps/web-giddh/src/app/settings/tags/tags.component.ts

```diff
@@ -33,42 +31,55 @@ export class SettingsTagsComponent implements OnInit, OnDestroy {
     public commonLocaleData: any = {};
 
     constructor(
-        private store: Store<AppState>,
-        private settingsTagsActions: SettingsTagActions
+        private settingsTagService: SettingsTagService,
+        private toaster: ToasterService
     ) {
     }
 
     public ngOnInit() {
-        this.store.pipe(select(state => state.settings.isGetAllTagsInProcess), takeUntil(this.destroyed$)).subscribe(response => {
-            this.isLoading = response;
-        });
+        this.getTags();
+    }
 
-        this.tags$ = this.store.pipe(select(createSelector([(state: AppState) => state.settings.tags], (tags) => {
-            if (tags && tags.length) {
-                map(tags, (tag) => {
+    public getTags() {
+        this.tags = [];
+        this.tagsBackup = null;
+        this.isLoading = true;
+        this.settingsTagService.GetAllTags().pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.status === "success" && response?.body?.length > 0) {
+                map(response?.body, (tag) => {
                     tag.uniqueName = tag.name;
                 });
-                let tagsData = orderBy(tags, 'name');
+                let tagsData = orderBy(response?.body, 'name');
+                this.tags = cloneDeep(tagsData);
                 this.tagsBackup = cloneDeep(tagsData);
-                return tagsData;
-            } else {
-                this.tagsBackup = null;
-                return null;
             }
-        })), takeUntil(this.destroyed$));
-    }
-
-    public getTags() {
-        this.store.dispatch(this.settingsTagsActions.GetALLTags());
+            this.isLoading = false;
+        });
     }
 
     public createTag(tag: TagRequest) {
-        this.store.dispatch(this.settingsTagsActions.CreateTag(tag));
+        this.settingsTagService.CreateTag(tag).pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            this.toaster.clearAllToaster();
```

Comment:
Consider moving this block to a new method that will take the `response` and success message as argument for `create`, `update` and `delete` tags scenarios